### PR TITLE
Add util_linux and update glib

### PR DESCRIPTION
**Pull request by flyingP0tat0**
Second try for glib update.

This time, mount functionality (and many useful things more), is provided by [util-linux](https://en.wikipedia.org/wiki/Util-linux). One could even think about making this a core package...

Depends on #1045.

### CODE REVIEW

**Comment by jam7** [comment]
`crew install util_linux` shows following error.
```
  CC       sys-utils/nsenter.o
  CC       sys-utils/setpriv.o
sys-utils/setpriv.c:21:20: fatal error: cap-ng.h: No such file or directory
 #include <cap-ng.h>
                    ^
```
BTW, why don't you just add `--disable-libmount` flat to glib.rb?  It's much easier.   We don't have ability to use libmount through previous glib.

**Comment by flyingP0tat0** [comment]
@jam7 I added `libcap_ng`, does it now work on your machine?
I did not encounter this error...

File: packages/glib.rb [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
-  source_url 'https://ftp.gnome.org/pub/gnome/sources/glib/2.40/glib-2.40.2.tar.xz'
-  source_sha256 'e8ff8af2950897e805408480c454c415d1eade4e670ec5fb507f5e5853726c7a'
+  homepage 'https://developer.gnome.org/glib'
+  version '2.52.3'
+  source_url 'https://ftp.gnome.org/pub/gnome/sources/glib/2.52/glib-2.52.3.tar.xz'
+  source_sha256 '25ee7635a7c0fcd4ec91cbc3ae07c7f8f5ce621d8183511f414ded09e7e4e128'
 
   depends_on 'libffi'
   depends_on 'gettext'
+  depends_on 'util_linux'
```
Comment (uberhacker) [inline packages/glib.rb]: Is `depends_on 'pcre'` needed?

**Comment by thedamian** [comment]
Anything that has an ADD I usually approve after some basic testing, but since you've mixed an add and a update and @jam7  found a possible bug and a great suggestion I'm going to wait to test it myself or to see if jam7 find that it's fixed now.

File: packages/glib.rb (hunk 1 above)
Comment (flyingP0tat0) [inline packages/glib.rb]: Yes it is. Won't build without.

Comment (flyingP0tat0) [inline packages/glib.rb]: Oh, wait, you meant if this needs to be added?
In this case, no, `with-pcre` says to use glib-provided pcre.

**Comment by flyingP0tat0** [comment]
@jam7 Could you test this?

**Comment by jam7** [comment]
Hmm, strange.  I received email that @flyingP0tat0 asking about the situation of this PR, but I cannot find that message on this PR.

Anyway, I'm waiting for fixes of #1045 before test this PR since this dpends on #1045.

File: packages/util_linux.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  homepage 'https://www.kernel.org'
+  version '2.30'
+  source_url 'https://www.kernel.org/pub/linux/utils/util-linux/v2.30/util-linux-2.30.tar.xz'
+  source_sha256 'c208a4ff6906cb7f57940aa5bc3a6eed146e50a7cc0a092f52ef2ab65057a08d'
+
+  depends_on 'python3'
+  depends_on 'vdev'
+
+  def self.build
+    system "./configure --prefix=/usr/local"
```
Comment (uberhacker) [inline packages/util_linux.rb]: `--prefix=#{CREW_PREFIX}` ?

**Comment by flyingP0tat0** [comment]
@jam7 This sould also work now.

File: packages/libcap_ng.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  source_url 'https://people.redhat.com/sgrubb/libcap-ng/libcap-ng-0.7.8.tar.gz'
+  source_sha256 'c21af997445cd4107a55d386f955c5ea6f6e96ead693e9151277c0ab5f97d05f'
+
+  def self.build
+    system "./configure"
+    system "make"
+  end
+
+  def self.install
+ 	  system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
```
Comment (jam7) [inline packages/libcap_ng.rb]: TAB here.

File: packages/util_linux.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+class Util_linux < Package
+  description 'essential linux tools'
+  homepage 'https://www.kernel.org'
+  version '2.30'
+  source_url 'https://www.kernel.org/pub/linux/utils/util-linux/v2.30/util-linux-2.30.tar.xz'
+  source_sha256 'c208a4ff6906cb7f57940aa5bc3a6eed146e50a7cc0a092f52ef2ab65057a08d'
+
+  depends_on 'python3'
+  depends_on 'vdev'
```
Comment (jam7) [inline packages/util_linux.rb]: Need dependency to `libcap_ag`?

When I compile this, this crerates following files.  It doesn't use python3 if python27 is installed.  You may need to specify python path at configure option to force to use python3.  Also bash_completion stuff conflicts with bash_completion.rb.  I don't care about latter.  It's just FYI.
```
drwxr-xr-x chronos/chronos       0 2017-09-18 03:34 usr/local/lib/python2.7/
drwxr-xr-x chronos/chronos       0 2017-09-18 03:34 usr/local/lib/python2.7/site-packages/
drwxr-xr-x chronos/chronos       0 2017-09-18 03:34 usr/local/lib/python2.7/site-packages/libmount/
-rwxr-xr-x chronos/chronos  255832 2017-09-18 03:34 usr/local/lib/python2.7/site-packages/libmount/pylibmount.so
-rw-r--r-- chronos/chronos      27 2017-09-18 03:34 usr/local/lib/python2.7/site-packages/libmount/__init__.py
-rw-r--r-- chronos/chronos     187 2017-09-18 03:34 usr/local/lib/python2.7/site-packages/libmount/__init__.pyo
-rw-r--r-- chronos/chronos     187 2017-09-18 03:34 usr/local/lib/python2.7/site-packages/libmount/__init__.pyc
-rwxr-xr-x chronos/chronos    1063 2017-09-18 03:34 usr/local/lib/python2.7/site-packages/libmount/pylibmount.la
...
-rw-r--r-- chronos/chronos    2591 2017-09-18 03:34 usr/local/share/bash-completion/completions/setterm
-rw-r--r-- chronos/chronos    1249 2017-09-18 03:34 usr/local/share/bash-completion/completions/zramctl
-rw-r--r-- chronos/chronos    1304 2017-09-18 03:34 usr/local/share/bash-completion/completions/lsipc
-rw-r--r-- chronos/chronos     576 2017-09-18 03:34 usr/local/share/bash-completion/completions/ipcmk
-rw-r--r-- chronos/chronos     509 2017-09-18 03:34 usr/local/share/bash-completion/completions/colrm
-rw-r--r-- chronos/chronos     726 2017-09-18 03:34 usr/local/share/bash-completion/completions/blockdev
-rw-r--r-- chronos/chronos    1167 2017-09-18 03:34 usr/local/share/bash-completion/completions/lsns
-rw-r--r-- chronos/chronos     721 2017-09-18 03:34 usr/local/share/bash-completion/completions/fallocate
-rw-r--r-- chronos/chronos    1007 2017-09-18 03:34 usr/local/share/bash-completion/completions/hexdump
-rw-r--r-- chronos/chronos    1961 2017-09-18 03:34 usr/local/share/bash-completion/completions/lsblk
-rw-r--r-- chronos/chronos     874 2017-09-18 03:34 usr/local/share/bash-completion/completions/flock
...
```

Comment (flyingP0tat0) [reply packages/util_linux.rb]: Changed to python27.
I don't think that I can do something about the completion stuff.

**Comment by jam7** [comment]
Having following errors.  Need `depends_on 'libcap_ng'` I think.
```
$ crew install util_linux
...
  CC       sys-utils/setpriv.o
sys-utils/setpriv.c:21:20: fatal error: cap-ng.h: No such file or directory
 #include <cap-ng.h>
                    ^
compilation terminated.
make[2]: *** [Makefile:7701: sys-utils/setpriv.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/usr/local/tmp/crew/util-linux-2.30.tar.xz.dir/util-linux-2.30'
make[1]: *** [Makefile:11718: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/util-linux-2.30.tar.xz.dir/util-linux-2.30'
make: *** [Makefile:4913: all] Error 2
```

**Comment by flyingP0tat0** [comment]
@jam7 Yes, that could be it. Can you try it with `libcap-ng` installed?

**Comment by jam7** [comment]
@flyingP0tat0 I tried it already as I explained at https://github.com/skycocker/chromebrew/pull/1050#discussion_r139389384.  ;)

**Comment by flyingP0tat0** [comment]
@jam7 Updated, does it now work?

**Comment by jam7** [comment]
Another problem I've found.  Sry, I didn't notice this yesterday.
```
$ crew install libcap_ng
...
checking for swig... no
configure: WARNING: "Swig not found - python bindings will not be made"
checking whether to create python bindings... testing
checking for python... no
checking for python2... no
checking for python3... no
checking for python3.3... no
checking for python3.2... no
checking for python3.1... no
checking for python3.0... no
checking for python2.7... no
checking for python2.6... no
checking for python2.5... no
checking for python2.4... no
checking for python2.3... no
checking for python2.2... no
checking for python2.1... no
checking for python2.0... no
configure: error: no suitable Python interpreter found
libcap_ng failed to build: `./configure` exited with 1
```

**Comment by flyingP0tat0** [comment]
@jam7 I added the `python3` dependency.

**Pull request merged**